### DISCUSSION TITLE
Loading the .pth file trained in the Isaac Gym environment

### DISCUSSION THREAD

**Discussion by famora2** [Toni-SM/skrl#19] [Ideas]
Hi, 

I really like this feature in skrl which enables simultaneous deploying of agents (https://skrl.readthedocs.io/en/latest/intro/examples.html). However, it would be great if skrl for example can also load the .pth file trained with Isaac Gym Library(https://github.com/NVIDIA-Omniverse/IsaacGymEnvs) for benchmark purposes. Currently, one gets the following errors: 

```

  File "ppo_ant.py", line 135, in <module>
    models_ppo["policy"].load("./ant_best.pth")
  File "/home/user/skrl/skrl/models/torch/base.py", line 302, in load
    self.load_state_dict(torch.load(path))
  File "/home/user/.local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 1483, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for Policy:
	Missing key(s) in state_dict: "log_std_parameter", "net.0.weight", "net.0.bias", "net.2.weight", "net.2.bias", "net.4.weight", "net.4.bias", "net.6.weight", "net.6.bias". 
	Unexpected key(s) in state_dict: "running_mean_std", "reward_mean_std", "model", "epoch", "optimizer", "frame", "last_mean_rewards", "env_state". 
```

**Comment by Toni-SM** [comment]
Hi @famora2 

The idea is a good one. Due to the incompatibility of the models implemented by both libraries, direct loading of checkpoints may not always be possible. Maybe an implementation that specifies the parameters' mapping could allow the migration. Hopefully, for the new version, this feature can be implemented

**Comment by Toni-SM** [comment]
Hi @famora2 

I had forgotten to continue this discussion.
With the current released version (v0.8.0) it is possible to load agents/models trained with other libraries inside skrl.

Refer to [Migrating external checkpoints](https://skrl.readthedocs.io/en/latest/intro/data.html#migrating-external-checkpoints) in docs for more details :)
